### ISSUE TITLE
Deploying twice leads to bizarre errors

### ISSUE THREAD

**Issue by alexlokshin** [camilb/prometheus-kubernetes#34]
When deployed over and over again, at a certain point errors like these surface up:

```
The Prometheus "k8s" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
configmap "prometheus-k8s-rules" created
The ServiceMonitor "alertmanager" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
The ServiceMonitor "kube-apiserver" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
The ServiceMonitor "kube-controller-manager" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
The ServiceMonitor "kube-dns" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
The ServiceMonitor "kube-scheduler" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
The ServiceMonitor "kube-state-metrics" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
The ServiceMonitor "kubelet" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
The ServiceMonitor "node-exporter" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
The ServiceMonitor "prometheus-operator" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
The ServiceMonitor "prometheus" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
```

**Comment by camilb** [comment]
@alexlokshin After running `./teardown`plese delete the custom resource definitions:

`kubectl get crd -o 'jsonpath={.items[*].metadata.name}' | xargs -n 1 | fgrep coreos.com | xargs kubectl delete crd` 

Then try deploying it again. Please let me know if this solves the issue.

**Comment by alexlokshin** [comment]
I added the command to teardown and ran the teardown script. Then deployed project a couple of times without any changes to code. Seems to deploy fine now.
 
Do we need to teardown every time before deploying any changes to project?

Another question: Did you remove Docker error count pane from the latest k8s-Usage dashboard? That was very useful for us.

Thank you for your help. Your project is awesome!

**Comment by camilb** [comment]
> Do we need to teardown every time before deploying any changes to project?

I never tried to apply the same configuration for `ServiceMonitor` (only updated rules, dashboards, etc) but seems to be a issue with Kubernetes itself: https://github.com/coreos/prometheus-operator/issues/699
I'll take a look on it.

> Another question: Did you remove Docker error count pane from the latest k8s-Usage dashboard? That was very useful for us.

Yes, I think it was removed at some moment to reduce the number of queries in the same dashboard  and avoid due to metrics changes in Prometheus 2.0. I'll test it soon and add it again

Thanks for appreciation! The hard work was done by CoreOS team, I'm just trying to simplify and automate some parts.

**Comment by camilb** [comment]
According to this https://github.com/kubernetes/kubernetes/issues/53379, the fix will arrive in Kubernetes 1.8.4

**Comment by camilb** [comment]
Can not reproduce it anymore in 1.8.4 
<img width="1280" alt="screen shot 2017-12-31 at 11 50 31" src="https://user-images.githubusercontent.com/2655869/34460769-113fe210-ee21-11e7-84ff-c3da08d646ae.png">
